packages/flutter/lib/utils/key_manager.dart

```diff
@@ -26,196 +26,163 @@ class KeyManager {
-    return hdKey.serializePublic(HDExtendedKeyVersion.xpub);
-  }
-
-  // getAddress
-  // returns the address of the account by alias
-  Future<String> getAddress(int index, {String? alias}) async {
-    HDKey pubKey = await _getHdKey(index, alias: alias);
-    return pubKey.encodeAddress();
+    final hd = _deriveHdKey(seed, index);
+    return hd;
   }
 
   Future<bool> _authWrapper() async {
     final didAuth = await _auth.authenticate(
       localizedReason: 'Please authenticate to access keystore',
     );
-    // if(didAuth == true) return true;
     return didAuth;
   }
+
   // _sign
   // requires verification
   // uses the hd wallet in the keystore to sign the message
-
-  Future<BigInt> _getPrivateKey(int index, {String? alias}) async {
+  Future<EthPrivateKey> _getPrivateKey(int index, {String? alias}) async {
     final isAuthenticated = await _authWrapper();
     if (isAuthenticated) {
       final hdKey = await _getHdKey(index, alias: alias);
-      final privateKey = hdKey.privKey;
-
-      // Ensure the privateKey is not null.
-      if (privateKey == null) {
-        throw Exception("Private key not found");
-      }
-
+      final privateKey = _deriveEthPrivKey(hdKey.privateKeyHex());
       return privateKey;
     } else {
       // Handle authentication failure here if needed.
       throw Exception("Authentication failed");
     }
   }
 
-  // _getMnemonic
-  // TODO:
-  // wrap with  auth
-
-  // signMessage
-  // external function that does the actual signing
-  Future<MsgSignature> signMessage(String message,
-      {int? index, String? alias}) async {
-    final privKey = await _getPrivateKey(index ?? 0, alias: alias);
-    Credentials credentials = EthPrivateKey.fromInt(privKey);
+  Future<String> _getMnemonic(String alias) async {
+    final isAuthenticated = await _authWrapper();
+    if (isAuthenticated) {
+      final mnemonic = await _getSecret(alias: ksNamespace);
+      if (mnemonic == null) throw "exportMnemonic: Not a Valid Wallet";
+      String decodedMnemonic = mnemonic;
+      return decodedMnemonic;
+    } else {
+      throw Exception("Authentication failed");
+    }
+  }
 
-    //hash the message
-    List<int> messageBytes = utf8.encode(message);
-    final hashMessage = sha256.convert(messageBytes);
-    final hashedMessage = Uint8List.fromList(hashMessage.bytes);
+  // generate account
+  // generates a new account based on hd
+  // stores the hdWallet on keystore
+  // returns an address
+  Future<String> generateAccount({String? alias}) async {
+    final mnemonic = _generate();
+    final seed = _recover(mnemonic, alias: alias);
+    return await _add(seed, 0, alias: alias);
+  }
 
-    //sign the message
-    final signedMessage = credentials.signToEcSignature(hashedMessage);
-    return signedMessage;
+Future<String> addAccount(String mnemonic, {String? alias}) async {
+  final seed = _recover(mnemonic, alias: alias);
+  return await _add(seed, 0, alias: alias);
+}
```

Comment:
This is wrong, add account should not take an mnemonic, it should take an index and an Alias.

Fetches the existing mnemonic, recovers the seed, adds the account access based on the specified index

And the add account function must be authenticated.